tests/zkevm/test_worst_bytecode.py

```diff
@@ -65,10 +63,43 @@ def test_worst_bytecode_single_opcode(
     The test is performed in the last block of the test, and the entire block gas limit is
     consumed by repeated opcode executions.
     """
-    # We use 100G gas limit to be able to deploy a large number of contracts in a single block,
-    # avoiding bloating the number of preparing blocks in the test.
-    env = Environment(gas_limit=100_000_000_000)
-    attack_gas_limit = Environment().gas_limit
+    # The attack gas limit is the gas limit which the target tx will use
+    # The test will scale the block gas limit to setup the contracts accordingly to be
+    # able to pay for the contract deposit. This has to take into account the 200 gas per byte,
+    # but also the quadratic memory expansion costs which have to be paid each time the
+    # memory is being setup
+    attack_gas_limit = 100_000_000
```

Comment:
```suggestion
    attack_gas_limit = Environment().gas_limit
```
This is the issue with the slow test, `Environment().gas_limit != env.gas_limit`, since the test is targeted to use the default block-gas-limit only for the attack (last transaction), and as much as possible for the contract deployment.